src/sql/workbench/browser/designer/designer.ts

```diff
@@ -934,11 +957,13 @@ export class Designer extends Disposable implements IThemable {
 		}
 	}
 
-	private getUIState(): DesignerUIState {
-		return {
-			activeContentTabId: this._contentTabbedPanel.activeTabId,
-			activeScriptTabId: this._scriptTabbedPannel.activeTabId
-		};
+	private saveUIState(): void {
+		if (this._input) {
+			this._input.designerUIState = {
```

Comment:
`this._input?.designerUIState = ...`

But also is an undefined input something you actually expect to happen? An editor not having a input doesn't seem like a valid scenario to me.

Reply:
this method gets called when input changes, we save the state to the current input, but for the first time, the current input will be undefined.

also, for assignment operation, the left side can't be optional.
![image](https://user-images.githubusercontent.com/13777222/157517323-6d15263e-3f53-4194-bffd-2e77956be1af.png)